pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"time"
+	"sync"
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+	
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
```

Comment:
I would keep the interface in `datarepair` that way the uses of it don't depend on the implementation code. This also should keep code cleaner when using it, because `datarepair` can refer to it just by `Queue`, rather than `queue.RepairQueue`. External packages can refer to it as `datarepair.Queue` which makes it nice to read.

Reply:
That should be fine, because it looks like `queue` doesn't need to import `datarepair`. Alternatively, if it does, then sometimes the dependency is injected from outside the package. But, if it really does cause circular-dependency we can leave as is for now.

Reply:
i'll leave it where it is for now, and we can move it later on if we can